### ISSUE TITLE
Problems when modifying the "WebRoot" and "ContentRoot"

### ISSUE THREAD

**Issue by gordon-matt** [peachpiecompiler/peachpie#187]
Modifying the `WebRoot` and/or `ContentRoot` is necessary to prevent HTTP 404 errors for js and css, etc.. but when doing so, it breaks normal use of ASP.NET Core, so we can't use `wwwroot` folder any longer. Example:

```
public class Program
{
    public static void Main(string[] args)
    {
        // TODO: How can we set root for PHP as different to root for MVC?
        // Meaning: wwwroot as usual for MVC and ../Website for PHP....
        var root = Path.GetDirectoryName(Directory.GetCurrentDirectory()) + "/Website";
        BuildWebHost(root, args).Run();
    }

    public static IWebHost BuildWebHost(string root, string[] args) =>
        WebHost.CreateDefaultBuilder(args)
            .UseStartup<Startup>()
            //.UseWebRoot(root) // Doing this fixes PHP, but breaks MVC
            //.UseContentRoot(root) // Doing this fixes PHP, but breaks MVC
            .Build();
}
```

**Comment by jakubmisek** [comment]
PHP root is set to WebRoot primarily. We can probably make an additional option for that, if it differs from Web Root that MVC needs.

**Comment by gordon-matt** [comment]
That would be great. Otherwise, I can't get them working together.

**Comment by gordon-matt** [comment]
@jakubmisek Saw your comment, here: https://github.com/gordon-matt/peachpie-responsive-file-manager/commit/050b1686ba0d6132661dbe61e03dfa7dc35adb44. Thanks, but the `RootPath` property on `PhpRequestOptions` does not yet exist. I assume the fix will be in the next nightly release, so I will wait for that.

**Comment by gordon-matt** [comment]
@jakubmisek I noticed you've now released it on NuGet. Thanks. I have used as follows:

```
app.UsePhp(new PhpRequestOptions(scriptAssemblyName: "ResponsiveFileManager")
{
    RootPath = Path.GetDirectoryName(Directory.GetCurrentDirectory()) + "/Website"
});
```

I assume that is correct, but I cannot yet confirm, because I am getting a new error which I will open a different issue for. However, I think either `RootPath` is not working properly or more likely I have made a mistake in setting it to the above, because I still see a 404 message for a .js file just before I get the new error (which prevents the php page from loading). Please confirm if I have used `RootPath` correctly.

**Comment by jakubmisek** [comment]
@gordon-matt are you sure `GetCurrentDirectory` returns an expected path? Note in debug it points to something like `/bin/debug/netcoreapp2.0/`

**Comment by gordon-matt** [comment]
While debugging, I added `Path.GetDirectoryName(Directory.GetCurrentDirectory()) + "\\Website"` to the Watch window and it produced the following: `D:\Source\GitHub\peachpie-responsive-file-manager\Website`. This is correct, yet I get 404s for all css and js

**Comment by jakubmisek** [comment]
@gordon-matt static files are handled by asp.net core (see `app.UseStaticFiles();`)  and it uses `ContentPath` property as the root path. You have to uncomment `.UseContentRoot(<path to content files>)`. It also have some configuration if you need to have it customized I guess : )

**Comment by gordon-matt** [comment]
@jakubmisek  I think you are misunderstanding. The problem is not with js and css files that relate to my ASP.NET Core site (those work fine, as they should). Rather, it's the css and js files that need to be loaded by the file manager. Before `RootPath`, when I changed the `WebRoot` and `ContentRoot`, then the file manager could load the css and js files without any problem at all. But then MVC could not load anything, because it could not look in **wwwroot** anymore. That's why you now have the `RootPath`, right? Well, it isn't working.

**Comment by jakubmisek** [comment]
Got it. New `RootPath` option might not get used properly, latest build CI00855 fixes that.

**Comment by gordon-matt** [comment]
@jakubmisek Just updated all references to use latest build (specific version number as you mentioned above).. did `dotnet restore` and rebuild.. no difference. I have updated [peachpie-responsive-file-manager](https://github.com/gordon-matt/peachpie-responsive-file-manager). Might be easier for you to diagnose the problem if you get the latest of that. Just set the `WebApplication` project to be the default (obviously) and browse to http://localhost:6107/iframe.

**Comment by jakubmisek** [comment]
Those files are being read by PHP code or by web browser?

**Comment by jakubmisek** [comment]
@gordon-matt it won't load compiled `ResponsiveFileManager.dll` during runtime, assuming it is not in current directory so .NET won't find it :/ 

```
FileNotFoundException: Could not load file or assembly 'ResponsiveFileManager, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
```

**Comment by gordon-matt** [comment]
@jakubmisek , I've never seen that **FileNotFoundException** before. After `dotnet restore`, did you "Reload Project" for the "Website" project? Otherwise, I don't know what else it could be.

**Comment by gordon-matt** [comment]
@jakubmisek 

There must be something here that I am not really understanding well, because I just tried setting the **WebRoot** and **ContentRoot** as before:

```
public class Program
{
	public static void Main(string[] args)
	{
		var root = Path.GetDirectoryName(Directory.GetCurrentDirectory()) + "/Website";
		BuildWebHost(root, args).Run();
	}

	public static IWebHost BuildWebHost(string root, string[] args) =>
		WebHost.CreateDefaultBuilder(args)
			.UseStartup<Startup>()
			.UseWebRoot(root) // Doing this fixes PHP, but breaks MVC
			.UseContentRoot(root) // Doing this fixes PHP, but breaks MVC
			.Build();
}
```

and the result is **MVC** can't find views, etc.. but when I go directly to the **PHP** file (http://localhost:6107/filemanager/dialog.php), then it shows everything just fine (loads the **JS** and **CSS**, etc).

Perhaps is it the case that ASP.NET Core still loads these static files, even though they're part of the PHP page? So does that mean I would need to split up the PHP and its assets so that only the PHP files are in the "Website" project but all of its assets (CSS, JS, images, etc) are moved to the Web App's **wwwroot** folder and then modify the **.php** files to change the `<scipt` and `<link` tags to point to the new location?

**Comment by gordon-matt** [comment]
@jakubmisek 

OK I guessed right.. and this problem is solved. Basically, I created a folder with the exact same name (**filemanager**) inside **wwwroot** and then moved all static files from **Website\filemanager** to **WebApplication\wwwroot\filemanager**, while still leaving the **.php** files where they were. This seems to be working well. I'm not sure if that's the best solution in your opinion, but it's the only thing that works for now.

**Comment by jakubmisek** [comment]
Yes, or you can look on .UseStaticFiles(). It has options, you basically need to provide two FileProvider s.

**Comment by gordon-matt** [comment]
@jakubmisek Thanks for the tip; I will look into that.

**Comment by jakubmisek** [comment]
@gordon-matt according to this == https://stackoverflow.com/questions/41021715/how-can-you-use-multiple-directories-for-static-files-in-an-aspnet-core-app == you can use UseStaticFiles twice for both directories.

**Comment by gordon-matt** [comment]
Appreciate that, thanks.
